src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+
+    warning_detection_method: BaseDriftDetector
+        Warning Detection method.
+
+    is_background_learner: bool
+        True if the tree is a background learner.
+
+    Notes
+    -----
+    Inner class that represents a single tree member of the forest.
+    Contains analysis information, such as the numberOfDriftsDetected.
+    """
+    def __init__(self,
+                 index_original,
+                 estimator: RegressionHoeffdingTree,
+                 instances_seen,
+                 drift_detection_method: BaseDriftDetector,
+                 warning_detection_method: BaseDriftDetector,
+                 is_background_learner):
+        self.index_original = index_original
+        self.estimator = estimator
+        self.created_on = instances_seen
+        self.is_background_learner = is_background_learner
+        self.evaluator_method = RegressionMeasurements
+
+        # Drift and warning
+        self.drift_detection_method = drift_detection_method
+        self.warning_detection_method = warning_detection_method
+
+        self.last_drift_on = 0
+        self.last_warning_on = 0
+        self.nb_drifts_detected = 0
+        self.nb_warnings_detected = 0
+
+        self.drift_detection = None
+        self.warning_detection = None
+        self.background_learner = None
+        self._use_drift_detector = False
+        self._use_background_learner = False
+        
+        self.evaluator = self.evaluator_method()
+
+        # Initialize drift and warning detectors
+        if drift_detection_method is not None:
+            self._use_drift_detector = True
+            self.drift_detection = deepcopy(drift_detection_method)
+
+        if warning_detection_method is not None:
+            self._use_background_learner = True
+            self.warning_detection = deepcopy(warning_detection_method)
+
+    def reset(self, instances_seen):
+        if self._use_background_learner and self.background_learner is not None:
+            self.estimator = self.background_learner.estimator
+            self.warning_detection = self.background_learner.warning_detection
+            self.drift_detection = self.background_learner.drift_detection
+            self.evaluator_method = self.background_learner.evaluator_method
+            self.created_on = self.background_learner.created_on
+            self.background_learner = None
+        else:
+            self.estimator.reset()
+            self.created_on = instances_seen
+            self.drift_detection.reset()
+        self.evaluator = self.evaluator_method()
+
+    def partial_fit(self, X, y, sample_weight, instances_seen):
+        self.estimator.partial_fit(X, y, sample_weight=sample_weight)
+
+        if self.background_learner:
+            self.background_learner.estimator.partial_fit(X, y, sample_weight=sample_weight)
+
+        if self._use_drift_detector and not self.is_background_learner:
+            correctly_classifies = self.estimator.predict(X) == y
```

Comment:
This could be misleading for regression tasks. Would an acceptable amount of error be preferred?

Reply:
This is not correct. The prediction should be used directly as input for `ADWIN`. This is also relevant since not all drift detection methods work on continuous data.

Reply:
Indeed, I did not really think about it. I now just add the predicted value, as suggested.

Reply:
Could you check that @hmgomes and @jacobmontiel?

Reply:
It seems correct to me now. Thanks!